### PR TITLE
x86: support XOP horizontal add/sub instructions

### PR DESCRIPTION
**Pull request by batuzovk**
Support decoding of XOP instructions from horizontal add/sub group. Previously there were only IDs for them, and empty decoding tables.

Code sample for testing purposes:

```
_start:
    vphaddbw %xmm3, %xmm1
    vphaddbw 0x123(%rbx,%rcx,4), %xmm1
    vphaddbd %xmm3, %xmm1
    vphaddbd 0x123(%rbx,%rcx,4), %xmm1
    vphaddbq %xmm3, %xmm1
    vphaddbq 0x123(%rbx,%rcx,4), %xmm1
    vphaddwd %xmm3, %xmm1
    vphaddwd 0x123(%rbx,%rcx,4), %xmm1
    vphaddwq %xmm3, %xmm1
    vphaddwq 0x123(%rbx,%rcx,4), %xmm1
    vphadddq %xmm3, %xmm1
    vphadddq 0x123(%rbx,%rcx,4), %xmm1
    vphaddubw %xmm3, %xmm1
    vphaddubw 0x123(%rbx,%rcx,4), %xmm1
    vphaddubd %xmm3, %xmm1
    vphaddubd 0x123(%rbx,%rcx,4), %xmm1
    vphaddubq %xmm3, %xmm1
    vphaddubq 0x123(%rbx,%rcx,4), %xmm1
    vphadduwd %xmm3, %xmm1
    vphadduwd 0x123(%rbx,%rcx,4), %xmm1
    vphadduwq %xmm3, %xmm1
    vphadduwq 0x123(%rbx,%rcx,4), %xmm1
    vphaddudq %xmm3, %xmm1
    vphaddudq 0x123(%rbx,%rcx,4), %xmm1
    vphsubbw %xmm3, %xmm1
    vphsubbw 0x123(%rbx,%rcx,4), %xmm1
    vphsubwd %xmm3, %xmm1
    vphsubwd 0x123(%rbx,%rcx,4), %xmm1
    vphsubdq %xmm3, %xmm1
    vphsubdq 0x123(%rbx,%rcx,4), %xmm1
```

With this change Dyninst text disassemble output (from examples/disassemble) matches source assembler and output of objdump.

There is a duplication of `vextab = true;` between this PR and #1752 . Once one of them is merged I'll remove this line in the other.

### CODE REVIEW

**Comment by hainest** [comment]
>There is a duplication of vextab = true; between this PR and https://github.com/dyninst/dyninst/pull/1752 . Once one of them is merged I'll remove this line in the other.

I'll wait until that one is merged before testing this one.

**Comment by batuzovk** [comment]
The code was rebased and updated

**Comment by hainest** [comment]
No test regressions.

**Review by hainest — APPROVED** [review]

**Pull request merged**
